src/main/java/org/sonarlint/intellij/util/SonarLintUtils.java

```diff
@@ -108,12 +112,40 @@ public void run() {
     return false;
   }
 
+  public static boolean shouldAnalyzeAutomatically(@Nullable VirtualFile file, @Nullable Module module) {
+    if (!shouldAnalyze(file, module)) {
+      return false;
+    }
+
+    // file and module not null here
+    if ("java".equalsIgnoreCase(file.getFileType().getDefaultExtension()) && !isSource(file, module)) {
```

Comment:
![BLOCKER](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-blocker.png) NullPointerException might be thrown as 'file' is nullable here [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://dory.sonarsource.com/coding_rules#rule_key=squid%3AS2259)